src/portable/nordic/nrf5x/dcd_nrf5x.c

```diff
@@ -82,17 +82,8 @@ typedef struct
 
 /*static*/ struct
 {
-  struct
-  {
-    uint8_t* buffer;
-    uint16_t total_len;
-    volatile uint16_t actual_len;
-
-    uint8_t  dir;
-  }control;
-
-  // Non control: 7 endpoints IN & OUT (offset 1)
-  nom_xfer_t xfer[7][2];
+  // All 8 endpoints including control IN & OUT (offset 1)
+  nom_xfer_t xfer[8][2];
```

Comment:
Yeah :+1: , I also want to merge them together, but always be stuck to something else :(